Bugfixes

tfheen opened the pull request:
A small selection of bugfixes. I don't believe anything in here is controversial in any way.

tfheen committed:
Only pass content to status for delete_domain

self is an implied parameter.  Avoids backtrace.

tfheen committed:
Avoid infinite loop in backend.healtcheck()

Call __getattr__ directly, else we end up calling ourselves which ends
in tears.

tfheen committed:
Comment is valid for conditions, make that available

tfheen committed:
Allow retrieving the backends from a director

tfheen committed:
Use the same keyword name in create_header and update_header

tfheen committed:
Make it possible to use the same kwargs for update_{condition,director,header} as their create_ counterparts

tfheen committed:
Add support for setting SNI host name

copyconstruct commented:
@tfheen Would it be possible to rebase off the latest master? Would be happy to merge then.

tfheen commented:
Sure, that should be possible. We have a bunch of other fixes in the pipe too, I'll get you PRs for those too.

copyconstruct commented:
Thanks! Appreciate it.
On Fri, Apr 7, 2017 at 12:52 AM Tollef Fog Heen <notifications@github.com>
wrote:

> Sure, that should be possible. We have a bunch of other fixes in the pipe
> too, I'll get you PRs for those too.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/zebrafishlabs/fastly-python/pull/14#issuecomment-292467601>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AY_nqKvE4rbo5dbctPGkYZZXf32Gn2vbks5rtetFgaJpZM4IseCU>
> .
>

tfheen committed:
Return None if healthcheck is None, rather than trying to retrieve one called None from the API

tfheen committed:
Add support for setting ssl_cert_hostname

tfheen committed:
Pass safe='' to urllib.quote

Names can (often) contain slashes, so we need to make sure those are
encoded as well.

tfheen commented:
@copyconstruct: sorry it took a while to get around to this. Let me know if you would rather have this split out into a bunch of PRs, I can do that if you prefer.

tfheen committed:
Treat healthcheck == None and healthcheck == '' the same

Fastly's API doesn't properly differentiate between
healthcheck-as-a-string and as healthcheck-as-an-object, which means
you will get healthcheck == '' back when it says "(none)" in the UI.

tfheen committed:
Simplify getattr call somewhat, "" is false in python

tfheen commented on fastly/__init__.py:
```diff
@@ -1132,11 +1142,15 @@ class FastlyBackend(FastlyObject, IServiceVersionObject):
 	@property
 	def healthcheck(self):
-		return self._conn.get_healthcheck(self.service_id, self.version, self.healthcheck)
+                if self.__getattr__('healthcheck') is None or self.__getattr__('healthcheck') == '':
```
Indeed, fixed. A bit damaged by writing ruby. :-)

tfheen commented:
@copyconstruct, any chance for a new review here (or possibly a merge)? Happy to make changes if you want those, but it'd be great to see movement on the project too.

copyconstruct commented on fastly/__init__.py:
```diff
@@ -1132,11 +1142,15 @@ class FastlyBackend(FastlyObject, IServiceVersionObject):
 	@property
 	def healthcheck(self):
-		return self._conn.get_healthcheck(self.service_id, self.version, self.healthcheck)
+                if self.__getattr__('healthcheck') is None or self.__getattr__('healthcheck') == '':
```
Could it be possible you didn't push up the change? Because it still shows the old diff to me.

tfheen replied:
Yes, you're right, it seems like I only added it to my master branch, not the bugfixes branch.

copyconstruct replied:
Cool, thanks! have merged.

tfheen commented:
Thanks for merging, much appreciated!
